From: Gwenhael Goavec-Merou <gwenhael.goavec-merou@trabucayre.com>
Date: Mon, 10 Sep 2018 15:52:09 +0200
Subject: [PATCH] libusbgx: fix build with glibc-2.28 since <sys/sysmacros.h>
 is no more included by <sys/types.h>

Signed-off-by: Sid Spry R030t1@gmail.com
Signed-off-by: Gwenhael Goavec-Merou gwenhael.goavec-merou@trabucayre.com
[Copy sign-offs from pull requst to commit msg]
Signed-off-by: Krzysztof Opasiak <k.opasiak@samsung.com>
---
 examples/gadget-acm-ecm.c              | 1 +
 examples/gadget-import.c               | 1 +
 examples/gadget-ms.c                   | 1 +
 examples/show-gadgets.c                | 1 +
 examples/show-udcs.c                   | 1 +
 include/usbg/usbg_internal_libconfig.h | 1 +
 6 files changed, 6 insertions(+)

diff --git a/examples/gadget-acm-ecm.c b/examples/gadget-acm-ecm.c
index 1c5e2ca5161c..29360da15811 100644
--- a/examples/gadget-acm-ecm.c
+++ b/examples/gadget-acm-ecm.c
@@ -20,6 +20,7 @@
 
 #include <errno.h>
 #include <stdio.h>
+#include <sys/sysmacros.h>
 #include <linux/usb/ch9.h>
 #include <usbg/usbg.h>
 
diff --git a/examples/gadget-import.c b/examples/gadget-import.c
index e684fdb861e8..63df4499926b 100644
--- a/examples/gadget-import.c
+++ b/examples/gadget-import.c
@@ -25,6 +25,7 @@
 #include <errno.h>
 #include <string.h>
 #include <stdio.h>
+#include <sys/sysmacros.h>
 #include <usbg/usbg.h>
 
 int main(int argc, char **argv)
diff --git a/examples/gadget-ms.c b/examples/gadget-ms.c
index 478c37097397..a5c668187a96 100644
--- a/examples/gadget-ms.c
+++ b/examples/gadget-ms.c
@@ -23,6 +23,7 @@
 
 #include <errno.h>
 #include <stdio.h>
+#include <sys/sysmacros.h>
 #include <linux/usb/ch9.h>
 #include <usbg/usbg.h>
 #include <usbg/function/ms.h>
diff --git a/examples/show-gadgets.c b/examples/show-gadgets.c
index 707d4488d16b..a2a21c883b27 100644
--- a/examples/show-gadgets.c
+++ b/examples/show-gadgets.c
@@ -21,6 +21,7 @@
 #include <errno.h>
 #include <stdio.h>
 #include <string.h>
+#include <sys/sysmacros.h>
 #include <netinet/ether.h>
 #include <usbg/usbg.h>
 #include <usbg/function/ms.h>
diff --git a/examples/show-udcs.c b/examples/show-udcs.c
index 66e950f7bf85..2f5cc458ebb1 100644
--- a/examples/show-udcs.c
+++ b/examples/show-udcs.c
@@ -23,6 +23,7 @@
 
 #include <errno.h>
 #include <stdio.h>
+#include <sys/sysmacros.h>
 #include <usbg/usbg.h>
 
 int main(void)
diff --git a/include/usbg/usbg_internal_libconfig.h b/include/usbg/usbg_internal_libconfig.h
index ac51758b3d6c..3fa55c0b544b 100644
--- a/include/usbg/usbg_internal_libconfig.h
+++ b/include/usbg/usbg_internal_libconfig.h
@@ -12,6 +12,7 @@
 #ifndef USBG_INTERNAL_LIBCONFIG_H
 #define USBG_INTERNAL_LIBCONFIG_H
 
+#include <sys/sysmacros.h>
 #include <libconfig.h>
 #ifdef __cplusplus
 extern "C" {
